src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonArray.IList.cs

```diff
@@ -114,6 +116,116 @@ public void RemoveAt(int index)
             DetachParent(item);
         }
 
+        /// <summary>
+        ///   Removes all the elements that match the conditions defined by the specified predicate.
+        /// </summary>
+        /// <param name="match">The predicate that defines the conditions of the elements to remove.</param>
+        /// <returns>The number of elements removed from the <see cref="JsonArray"/>.</returns>
+        /// <exception cref="ArgumentNullException">
+        ///   <paramref name="match"/> is <see langword="null"/>.
+        /// </exception>
+        public int RemoveAll(Func<JsonNode?, bool> match)
+        {
+            if (match == null)
+            {
+                ThrowHelper.ThrowArgumentNullException(nameof(match));
+            }
+
+            if (_list is not { } list)
```

Comment:
This check is not correct. `_list` stores the lazily initialized underlying list which must have been initialized for this operation. Use the `List` property instead.